Tests/CategoriesControllerTest.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using WebNetSample.Business.Abstract;
+using WebNetSample.Entity.Concrete;
+using WebNetSample.Tests.Configurations;
+using WebNetSample.WebNetMVC.Controllers;
+
+namespace WebNetSample.Tests;
+
+public class CategoriesControllerTest
+{
+    [Theory, AutoMoqData]
+    public void Index_Should_Return_As_Expected(List<Category> expected)
+    {
+        // Arrange
+        var productServiceMock = new Mock<ICategoryService>();
+
+        var sut = new CategoriesController(productServiceMock.Object);
+        productServiceMock.Setup(c => c.GetListAsync().Result).Returns(expected);
+
+        // Act
+        var actionResult = sut.Index();
+
+        // Assert
+        var okViewResult = actionResult.Result as ViewResult;
+        Assert.NotNull(okViewResult);
+
+        var model = okViewResult.Model as List<Category>;
+        Assert.NotNull(model);
+
+        var actual = model;
+
+        for (int i = 0; i < actual.Count; i++)
```

Comment:
If your `actual` length does not equal to `expected` you will have a failed test by bad configuration. To avoid that, please, add assert for expected and actual collections length. If it fails - test will stop its work